### DISCUSSION TITLE
Echarts transitive dependency - consuming 787KB

### DISCUSSION THREAD

**Discussion by vaughansharman** [appbaseio/reactivesearch#2182] [General]
Reactive search is pulling in echarts, which is not required (as far as I can see) to use the product. 

Is it possible to have a build that us not dependent on this module?

<img width="522" alt="Screen Shot 2023-02-16 at 17 35 26" src="https://user-images.githubusercontent.com/2912404/219443764-878e10b8-b181-45c0-9515-d2cdb48f5e27.png">

**Comment by siddharthlatest** [comment]
@vaughansharman Echarts is used by ReactiveChart components. If you aren't using chart components, it shouldn't factor into your search UI's bundle size. Let me know if you're seeing otherwise.

**Reply by kasvith** [reply]
@siddharthlatest im also having same issue with vite. Currently vite build includes everything like ESChart etc into bundle.

SB: https://stackblitz.com/edit/vitejs-vite-ru75t2?file=package.json
If you do a production build, it has around 2MB bundle

**Reply by kasvith** [reply]
I also tried direct imports, but still same result

**Reply by bietkul** [reply]
@kasvith We have fixed it for v4.x, v3.x fix is in progress.

**Reply by kasvith** [reply]
@bietkul cant we use 4.x with Elastic Servers? Is it only supported for appbase io?

**Reply by bietkul** [reply]
yes

**Reply by kasvith** [reply]
dang, we have an ES server, so we cant use 4.x then :(

**Reply by vaughansharman** [reply]
Nice one, will keep an eye out for the 3.x fix, ty!

**Comment by vaughansharman** [comment]
Yeah, i'm not using charts and it's being included in the bundle, using `create-react-app` and reactive search `"@appbaseio/reactivesearch": "^3.43.7"`

I am using the following components:

ReactiveList
ReactiveBase
ReactiveComponent
SelectedFilters
DataSearch
ToggleButton
CategorySearch
SingleDropdownList
ResultCard

**Comment by vaughansharman** [comment]
`yarn why echarts`

```
yarn why v1.22.19
warning ../package.json: No license field
[1/4] 🤔  Why do we have the module "echarts"...?
[2/4] 🚚  Initialising dependency graph...
[3/4] 🔍  Finding dependency...
[4/4] 🚡  Calculating file sizes...
=> Found "echarts@5.4.1"
info Reasons this module exists
   - "@appbaseio#reactivesearch" depends on it
   - Hoisted from "@appbaseio#reactivesearch#echarts"
info Disk size without dependencies: "43.54MB"
info Disk size with unique dependencies: "48.76MB"
info Disk size with transitive dependencies: "48.76MB"
info Number of shared dependencies: 2
✨  Done in 0.35s.
```

**Comment by bietkul** [comment]
This issue is fixed in `v3.43.8`.

**Reply by vaughansharman** [reply]
thanks, can confirm echarts is no longer showing up in my distributable

**Reply by kasvith** [reply]
Yes i confirm the same
